src/main/java/com/midaswebserver/midasweb/services/UserServiceImp.java

```diff
@@ -0,0 +1,264 @@
+package com.midaswebserver.midasweb.services;
+
+import com.midaswebserver.midasweb.models.User.Symbol;
+import com.midaswebserver.midasweb.models.User.User;
+import com.midaswebserver.midasweb.repositories.UserRepository;
+import jakarta.servlet.http.HttpServletRequest;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+
+/**
+ * @Author Aidan Scott
+ * @since 0.0.1
+ * UserService is the CRUD service for Users. the majority of user logic goes through the
+ * UserServiceImp
+ */
+@Service
+public class UserServiceImp implements UserService {
+    private static final Logger log = LoggerFactory.getLogger(UserServiceImp.class);
+    private final UserRepository userRepo;
+
+    /**
+     * Injected dependencies
+     *
+     * @param userRepo
+     */
+    public UserServiceImp(UserRepository userRepo) {
+        this.userRepo = userRepo;
+        log.debug("User Repository initialized in UserService");
```

Comment:
Probably not a very useful log message, as if the constructor doesn't complete the entire application will fail to start.